spec/schemas/nat-gateway.yaml

```diff
@@ -0,0 +1,40 @@
+InternetNatGateway:
+  description: |
+    An InternetNatGateway is a resource that provides internet access to a subnet
+    by mangling the source IP address of outgoing packets. This allows instances
+    in the subnet to access the internet while keeping their private IP addresses
+    hidden. The InternetNatGateway is used in conjunction with a RouteTable to
+    enable internet connectivity.
+  allOf:
+  - $ref: './resource.yaml#/UserResourceMetadata'
+  - type: object
+    description: |
+      An InternetNatGateway is a resource that provides internet access to a subnet
+      by mangling the source IP address of outgoing packets. This allows instances
+      in the subnet to access the internet while keeping their private IP addresses
+      hidden. The InternetNatGateway is used in conjunction with a RouteTable to
+      enable internet connectivity.
+  required:
+  - spec
+  properties:
+    metadata:
+      $ref: './resource.yaml#/RegionalResourceMetadata'
+    spec:
+      $ref: './nat-gateway.yaml#/InternetNatGatewaySpec'
+    status:
+      $ref: './nat-gateway.yaml#/InternetNatGatewayStatus'
+
+InternetNatGatewaySpec:
+  type: object
+  description: |
+    The internet gateway will allocate a public IPv4 address.
+  required:
+  - publicIpRef
+  properties:
+    publicIpRef:
```

Comment:
Hi Vincent, discussing with Amedeo about this we think it would be essential to specify also a subnet.
The NATGW will be connected to at least one, and shouldn't be possible to create one without.
THe discussion we had was around if it is better a NIC to be associated to the NATGW or directly a SUBNET

Reply:
I changed it to be an NatGatewayInstance (L2) this way we can have the NIC and can later add a VRF type nat gateway (L3)...